src/components/HeadCellMenu/HeadCellMenu.tsx

```diff
@@ -50,8 +44,32 @@ const HeadCellMenu = ({
   }, [searchRelatedArgs, fieldId, qLibraryId]);
 
   useEffect(() => {
-    if (!openMenuDropdown) resetSelectionActionsEnabledStatus();
-  }, [openMenuDropdown, resetSelectionActionsEnabledStatus]);
+    if (!openMenuDropdown) {
+      resetSelectionActionsEnabledStatus();
+
+      if (selectionRelatedArgs?.model) {
+        selectionRelatedArgs.model
+          .getLayout()
+          .then((layout) => {
+            updateSelectionActionsEnabledStatus(layout as EngineAPI.IGenericHyperCubeLayout);
+          })
+          .catch((e) => console.error(e));
+      }
+    }
+  }, [
+    openMenuDropdown,
+    resetSelectionActionsEnabledStatus,
+    updateSelectionActionsEnabledStatus,
```

Comment:
`updateSelectionActionsEnabledStatus` is not wrapped in a `useCallback` is it will be new reference on every render